packages/env/lib/init-config.js

```diff
@@ -280,19 +276,17 @@ USER root`;
 }
 
 /**
- * Checks the configured PHP version
- * against the minimum version supported by Xdebug
+ * Gets the Xdebug config based on the options in the config object.
  *
  * @param {WPConfig} config
- * @return {boolean} Whether the PHP version is supported by Xdebug
+ * @return {string} The Xdebug config -- can be an empty string when it's not used.
  */
-function checkXdebugPhpCompatibility( config ) {
-	// By default, an undefined phpVersion uses the version on the docker image,
-	// which is supported by Xdebug 3.
-	const phpCompatibility = true;
+function getXdebugConfig( config ) {
+	if ( config.xdebug === 'off' ) {
+		return '';
+	}
 
-	// If PHP version is defined
-	// ensure it meets the Xdebug minimum compatibility requirment.
+	// Throw an error if someone tries to use Xdebug with an unsupported PHP version.
 	if ( config.env.development.phpVersion ) {
```

Comment:
Technically this would break if `development` and `tests` had a different PHP version and one was incompatible.

Reply:
true, I'll update this